serveur_calcul_python/utilitaires/frontend_chart_data_processing.py

```
@@ -0,0 +1,394 @@
+        df_reg_uni = pd.DataFrame()
+        df_reg_uni[cf_db.db_column_converted_value] = x_values_chart
+        df_reg_uni[cf_db.db_column_reg_sets_id] = int(er_pert.ruleset_id)
+        reg_ids=er_pert.get_unique_reg_ids_using_land_use([cubf])
+        if len(reg_ids)!=1:
+            raise ValueError(f"Le CUBF {cubf} doit correspondre à un seul règlement")
+        reg_a_util:int= int(reg_ids[0])
+        df_reg_uni[cf_db.db_column_parking_regs_id] = reg_a_util
+        reglement:PR.ParkingRegulations = er_pert.get_reg_by_id(reg_a_util)
+        unites = reglement.reg_def[cf_db.db_column_parking_unit_id].unique().tolist()
+        if len(unites)>1 and unites[0]!= units:
+            raise ValueError("L'option graphique ne supporte pas plus d'une unité à l'heure actuelle")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)